// formik-material-ui-pickers/dist/formik-material-ui-pickers.es6.js
import { createElement } from 'react';
import { DatePicker, KeyboardDatePicker, TimePicker, KeyboardTimePicker, DateTimePicker, KeyboardDateTimePicker } from '@material-ui/pickers';
import { getIn } from 'formik';

/*! *****************************************************************************
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the Apache License, Version 2.0 (the "License"); you may not use
this file except in compliance with the License. You may obtain a copy of the
License at http://www.apache.org/licenses/LICENSE-2.0

THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
MERCHANTABLITY OR NON-INFRINGEMENT.

See the Apache Version 2.0 License for specific language governing permissions
and limitations under the License.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

function __rest(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
}

function fieldToDatePicker(_a) {
    var field = _a.field, _b = _a.form, isSubmitting = _b.isSubmitting, touched = _b.touched, errors = _b.errors, setFieldValue = _b.setFieldValue, setFieldError = _b.setFieldError, disabled = _a.disabled, props = __rest(_a, ["field", "form", "disabled"]);
    var fieldError = getIn(errors, field.name);
    var showError = getIn(touched, field.name) && !!fieldError;
    return __assign(__assign(__assign({}, props), field), { error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled != undefined ? disabled : isSubmitting, onChange: function (date) {
            setFieldValue(field.name, date);
        },
        onError: function (error) {
            if (error !== fieldError && !(error == '' && !fieldError)) {
                setFieldError(field.name, error ? String(error) : undefined);
            }
        } });
}
function DatePicker$1(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (createElement(DatePicker, __assign({}, fieldToDatePicker(props)), children));
}
DatePicker$1.displayName = 'FormikMaterialUIDatePicker';

function fieldToKeyboardDatePicker(_a) {
    var disabled = _a.disabled, field = _a.field, _b = _a.form, isSubmitting = _b.isSubmitting, touched = _b.touched, errors = _b.errors, setFieldValue = _b.setFieldValue, setFieldError = _b.setFieldError, props = __rest(_a, ["disabled", "field", "form"]);
    var fieldError = getIn(errors, field.name);
    var showError = getIn(touched, field.name) && !!fieldError;
    return __assign(__assign(__assign({}, props), field), { error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled != undefined ? disabled : isSubmitting, onChange: function (date) {
            setFieldValue(field.name, date);
        },
        onError: function (error) {
            if (error !== fieldError && !(error == '' && !fieldError)) {
                setFieldError(field.name, error ? String(error) : undefined);
            }
        } });
}
function KeyboardDatePicker$1(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (createElement(KeyboardDatePicker, __assign({}, fieldToKeyboardDatePicker(props)), children));
}
KeyboardDatePicker$1.displayName = 'FormikMaterialUIKeyboardDatePicker';

function fieldToTimePicker(_a) {
    var disabled = _a.disabled, field = _a.field, _b = _a.form, isSubmitting = _b.isSubmitting, touched = _b.touched, errors = _b.errors, setFieldValue = _b.setFieldValue, setFieldError = _b.setFieldError, props = __rest(_a, ["disabled", "field", "form"]);
    var fieldError = getIn(errors, field.name);
    var showError = getIn(touched, field.name) && !!fieldError;
    return __assign(__assign(__assign({}, props), field), { error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled != undefined ? disabled : isSubmitting, onChange: function (date) {
            setFieldValue(field.name, date);
        },
        onError: function (error) {
            if (error !== fieldError && !(error == '' && !fieldError)) {
                setFieldError(field.name, error ? String(error) : undefined);
            }
        } });
}
function TimePicker$1(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (createElement(TimePicker, __assign({}, fieldToTimePicker(props)), children));
}
TimePicker$1.displayName = 'FormikMaterialUITimePicker';

function fieldToKeyboardTimePicker(_a) {
    var disabled = _a.disabled, field = _a.field, _b = _a.form, isSubmitting = _b.isSubmitting, touched = _b.touched, errors = _b.errors, setFieldValue = _b.setFieldValue, setFieldError = _b.setFieldError, props = __rest(_a, ["disabled", "field", "form"]);
    var fieldError = getIn(errors, field.name);
    var showError = getIn(touched, field.name) && !!fieldError;
    return __assign(__assign(__assign({}, props), field), { error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled != undefined ? disabled : isSubmitting, onChange: function (date) {
            setFieldValue(field.name, date);
        },
        onError: function (error) {
            if (error !== fieldError && !(error == '' && !fieldError)) {
                setFieldError(field.name, error ? String(error) : undefined);
            }
        } });
}
function KeyboardTimePicker$1(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (createElement(KeyboardTimePicker, __assign({}, fieldToKeyboardTimePicker(props)), children));
}
KeyboardTimePicker$1.displayName = 'FormikMaterialUIKeyboardTimePicker';

function fieldToDateTimePicker(_a) {
    var disabled = _a.disabled, field = _a.field, _b = _a.form, isSubmitting = _b.isSubmitting, touched = _b.touched, errors = _b.errors, setFieldValue = _b.setFieldValue, setFieldError = _b.setFieldError, props = __rest(_a, ["disabled", "field", "form"]);
    var fieldError = getIn(errors, field.name);
    var showError = getIn(touched, field.name) && !!fieldError;
    return __assign(__assign(__assign({}, props), field), { error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled != undefined ? disabled : isSubmitting, onChange: function (date) {
            setFieldValue(field.name, date);
        },
        onError: function (error) {
            if (error !== fieldError && !(error == '' && !fieldError)) {
                setFieldError(field.name, error ? String(error) : undefined);
            }
        } });
}
function DateTimePicker$1(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (createElement(DateTimePicker, __assign({}, fieldToDateTimePicker(props)), children));
}
DateTimePicker$1.displayName = 'FormikMaterialUIDateTimePicker';

function fieldToKeyboardDateTimePicker(_a) {
    var disabled = _a.disabled, field = _a.field, _b = _a.form, isSubmitting = _b.isSubmitting, touched = _b.touched, errors = _b.errors, setFieldValue = _b.setFieldValue, setFieldError = _b.setFieldError, props = __rest(_a, ["disabled", "field", "form"]);
    var fieldError = getIn(errors, field.name);
    var showError = getIn(touched, field.name) && !!fieldError;
    return __assign(__assign(__assign({}, props), field), { error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled != undefined ? disabled : isSubmitting, onChange: function (date) {
            setFieldValue(field.name, date);
        },
        onError: function (error) {
            if (error !== fieldError && !(error == '' && !fieldError)) {
                setFieldError(field.name, error ? String(error) : undefined);
            }
        } });
}
function KeyboardDateTimePicker$1(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (createElement(KeyboardDateTimePicker, __assign({}, fieldToKeyboardDateTimePicker(props)), children));
}
KeyboardDateTimePicker$1.displayName = 'FormikMaterialUIKeyboardDateTimePicker';

export { fieldToDatePicker, DatePicker$1 as DatePicker, fieldToKeyboardDatePicker, KeyboardDatePicker$1 as KeyboardDatePicker, fieldToTimePicker, TimePicker$1 as TimePicker, fieldToKeyboardTimePicker, KeyboardTimePicker$1 as KeyboardTimePicker, fieldToDateTimePicker, DateTimePicker$1 as DateTimePicker, fieldToKeyboardDateTimePicker, KeyboardDateTimePicker$1 as KeyboardDateTimePicker };
